tests/conftest.py

```diff
@@ -10,16 +10,23 @@
 DATABASE = "testdb"
 
 
-def override_arnold_adapter():
+def override_arnold_adapter() -> ArnoldAdapter:
     """Function for overriding the arnold adapter dependency"""
+    mongo_client = MongoClient()
+    database = mongo_client[DATABASE]
+    return ArnoldAdapter(database.client, db_name=DATABASE)
 
+
+@pytest.fixture()
+def mock_adapter() -> ArnoldAdapter:
+    """Return a mock adapter for testing."""
     mongo_client = MongoClient()
     database = mongo_client[DATABASE]
     return ArnoldAdapter(database.client, db_name=DATABASE)
```

Comment:
So these are basically the same but one is a fixture while the other is not? Can mock_adapter just be `return override_arnold_adapter()`?

Reply:
The override arnold adapter is used to override a dependency in the fastapi testclient. So while the code is identical their function is not. I tested with letting it return one or the other but it breaks the tests. Will leave it as is for now.